src/main/ruby/truffleruby/core/string.rb

```diff
@@ -1097,6 +1097,82 @@ def rindex(sub, finish = undefined)
     nil
   end
 
+  def byteindex(str, start = undefined)
+    is_regex_pattern = Primitive.is_a?(str, Regexp)
+
+    if Primitive.undefined?(start)
+      start = 0
+    else
```

Comment:
It seems nicer to use a natural default value here, i.e. `def byteindex(str, start = 0)`, and the same for the method below. Then these 3 lines + the `end` can be removed.

Reply:
Sure thing